packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -315,18 +315,21 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
   @visibleForTesting
   void initMouseTracker([MouseTracker? tracker]) {
     _mouseTracker?.dispose();
-    _mouseTracker = tracker ?? MouseTracker();
+    _mouseTracker = tracker ?? MouseTracker((Offset position) {
+      final HitTestResult result = HitTestResult();
+      hitTest(result, position);
+      return result;
+    });
   }
 
   @override // from GestureBinding
   void dispatchEvent(PointerEvent event, HitTestResult? hitTestResult) {
     _mouseTracker!.updateWithEvent(
       event,
-      // Enter and exit events should be triggered with or without buttons
-      // pressed. When the button is pressed, normal hit test uses a cached
-      // result, but MouseTracker requires that the hit test is re-executed to
-      // update the hovering events.
-      () => (hitTestResult == null || event is PointerMoveEvent) ? renderView.hitTestMouseTrackers(event.position) : hitTestResult,
+      // When the button is pressed, normal hit test uses a cached result, but
+      // MouseTracker requires that the hit test is re-executed to update the
+      // hovering events.
+      useCachedHitTestResult(event) ? null : hitTestResult,
```

Comment:
I don't think it's up to the `MouseTracker` to decide, because the `MouseTracker` has no idea how the hit test result it received was generated.

I have a better idea: let the `GestureBinding` provide a method that always return the uncached hit test. Let me do this right now.

Reply:
> I don't think it's up to the MouseTracker to decide, because the MouseTracker has no idea how the hit test result it received was generated.

I was thinking, we could just define that the MouseTracker always wants the hitTestResult that was obtained when the pointer associated with the event first went down (this is the only result we cache in `_hitTests` anyways, right?), if such a result is available, or null, if it is not. If you pass in any other hitTestResult, you're in violation of the API contract. Then, the mouse tracker can decide if it wants to use that cached hitTestResult or not, depending on the current event type.

Reply:
The `MouseTracker` just doesn't want any cached result. It always needs the hit test result performed at this frame. The current plan (before the revert) is that the `MouseTracker` always receive the fresh result if available or null if not.